File: src/main/java/xyz/devvydont/smprpg/items/blueprints/sets/rosegold/RoseGoldDrillBase.java
package xyz.devvydont.smprpg.items.blueprints.sets.rosegold;

import xyz.devvydont.smprpg.attribute.AttributeWrapper;
import xyz.devvydont.smprpg.items.CustomItemType;
import xyz.devvydont.smprpg.items.ItemClassification;
import xyz.devvydont.smprpg.items.attribute.AdditiveAttributeEntry;
import xyz.devvydont.smprpg.items.attribute.AttributeEntry;
import xyz.devvydont.smprpg.items.base.CustomItemBlueprint;
import xyz.devvydont.smprpg.items.interfaces.IModularToolComponent;
import xyz.devvydont.smprpg.services.ItemService;

import java.util.Collection;
import java.util.List;

public class RoseGoldDrillBase extends CustomItemBlueprint implements IModularToolComponent {

    public static final String attrKey = "rose_gold_drill_base";

    public RoseGoldDrillBase(ItemService itemService, CustomItemType type) {
        super(itemService, type);
    }

    @Override
    public Collection<AttributeEntry> getAttributes() {
        return List.of(
                new AdditiveAttributeEntry(AttributeWrapper.MINING_FORTUNE, 105, getAttrKey()),
                new AdditiveAttributeEntry(AttributeWrapper.MINING_POWER, -1, getAttrKey())
        );
    }

    @Override
    public String getAttrKey() {
        return attrKey;
    }

    @Override
    public ItemClassification getItemClassification() {
        return ItemClassification.ITEM;
    }

    @Override
    public String getComponentPrefix() { return "Rose Gold"; }
}

File: src/main/java/xyz/devvydont/smprpg/items/blueprints/sets/rosegold/RoseGoldDrillBase.kt
package xyz.devvydont.smprpg.items.blueprints.sets.rosegold

import xyz.devvydont.smprpg.attribute.AttributeWrapper
import xyz.devvydont.smprpg.items.CustomItemType
import xyz.devvydont.smprpg.items.ItemClassification
import xyz.devvydont.smprpg.items.attribute.AdditiveAttributeEntry
import xyz.devvydont.smprpg.items.attribute.AttributeEntry
import xyz.devvydont.smprpg.items.interfaces.IModularToolComponent
import xyz.devvydont.smprpg.services.ItemService
import xyz.devvydont.smprpg.util.items.ToolStats

class RoseGoldDrillBase(itemService: ItemService, type: CustomItemType) : RoseGoldAttributeItem(itemService, type),
    IModularToolComponent {

    override val itemClassification: ItemClassification get() = ItemClassification.ITEM

    override fun getAttributes(): MutableCollection<AttributeEntry?> {
        return mutableListOf(
            AdditiveAttributeEntry(AttributeWrapper.MINING_FORTUNE, ToolStats.ROSE_GOLD.fortune * 1.5, ATTR_KEY),
            AdditiveAttributeEntry(AttributeWrapper.MINING_POWER, -1.0, ATTR_KEY)
        )
    }

    override fun getAttrKey(): String {
        return ATTR_KEY
    }

    companion object {
        const val ATTR_KEY: String = "rose_gold_drill_base"
    }
}
